build.rs

```diff
@@ -68,33 +68,77 @@ fn main() {
             .raw_line(format!("type {} = wgc::id::{};", old_name, new_name));
     }
 
-    // WGPUBindGroupEntry has fields that are id's which can be 0
+    // WGPUBindGroupEntry.{buffer, sampler, textureView} is nullable
```

Comment:
I'm not going to veto this now - but this seems kinda unsustainable long term - why do we need to redefine these structs completely?

Reply:
So, i looked into bindgen if it allows making struct fields optional instead of rewriting whole struct but was unsuccessful to find it.

Other thing we could do is instead of [generating](https://github.com/gfx-rs/wgpu-native/blob/91e14b8ffda0a5ec1d87fd27b7b9e8e62495720d/build.rs#L68) `WGPU[Type]` -> `[Type]Id` generate `WGPU[Type]` -> `Option<[Type]Id>` (which I think is what should have been done earlier) and do `unwrap`/`handle_error`s in each function calls, though don't know what the overhead of it would be. Then completely remove all struct rewrites.

Reply:
ack